server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java

```diff
@@ -163,101 +168,79 @@ private void processEvent(ShuffleDataFlushEvent event) {
-        if (storage == null) {
-          LOG.error("Storage selected is null and this should not happen. event: {}", event);
-          break;
-        }
+      Storage storage = storageManager.selectStorage(event);
+      if (storage == null) {
+        LOG.error("Storage selected is null and this should not happen. event: {}", event);
+        return true;
+      }
 
-        if (!storage.canWrite()) {
-          // todo: Could we add an interface supportPending for storageManager
-          //       to unify following logic of multiple different storage managers
-          if (storageManager instanceof MultiStorageManager) {
-            event.increaseRetryTimes();
-            ShuffleServerMetrics.incStorageRetryCounter(storage.getStorageHost());
-            continue;
-          } else {
-            // To avoid being re-pushed to pending queue and make the server too much pressure,
-            // it's better to drop directly.
-            if (event.isPended()) {
-              LOG.error("Drop this event directly due to already having entered pending queue. event: {}", event);
-              break;
-            }
-            event.increaseRetryTimes();
-            event.markPended();
-            addPendingEvents(event);
-            return;
+      if (!storage.canWrite()) {
+        // todo: Could we add an interface supportPending for storageManager
+        //       to unify following logic of multiple different storage managers
+        if (event.getRetryTimes() <= retryMax) {
+          if (event.isPended()) {
+            LOG.error("Drop this event directly due to already having entered pending queue. event: {}", event);
+            return true;
           }
-        }
-
-        String user = StringUtils.defaultString(
-            shuffleServer.getShuffleTaskManager().getUserByAppId(event.getAppId()),
-            StringUtils.EMPTY
-        );
-        CreateShuffleWriteHandlerRequest request = new CreateShuffleWriteHandlerRequest(
-            storageType,
-            event.getAppId(),
-            event.getShuffleId(),
-            event.getStartPartition(),
-            event.getEndPartition(),
-            storageBasePaths.toArray(new String[storageBasePaths.size()]),
-            shuffleServerId,
-            hadoopConf,
-            storageDataReplica,
-            user,
-            maxConcurrencyOfSingleOnePartition);
-        ShuffleWriteHandler handler = storage.getOrCreateWriteHandler(request);
-        writeSuccess = storageManager.write(storage, handler, event);
-        if (writeSuccess) {
-          updateCommittedBlockIds(event.getAppId(), event.getShuffleId(), blocks);
-          ShuffleServerMetrics.incStorageSuccessCounter(storage.getStorageHost());
-          break;
-        } else {
           event.increaseRetryTimes();
           ShuffleServerMetrics.incStorageRetryCounter(storage.getStorageHost());
+          event.markPended();
+          addPendingEvents(event);
```

Comment:
How about re-pushing to `flushEventHandler` queue? I think the `pending` queue could be removed, which is strange after this PR refactoring.

Reply:
Make sense.